hypex/dataset/datagenerator.py

```diff
@@ -0,0 +1,293 @@
+            is_treatment:
+                Number of treatments (info about 'treatment' 0 or 1). Defaults to 1
+            num_discrete_features:
+                Number of discrete columns correlating with outcome. Defaults to 1
+            num_targets:
+                Number of columns with target values. Defaults to 1
+            is_binary_target:
+                If outcome should be binary. Defaults to False
+            na_columns:
+                Column or list of columns with NA values. Defaults to None
+            na_step:
+                Number of period to make NaN (step of range). Defaults to 0
+        """
+        self.num_records = num_records
+        self.num_features = num_features
+        self.num_info_cols = num_info_cols
+        self.is_treatment = is_treatment
+        self.num_targets = num_targets
+        self.num_discrete_features = num_discrete_features
+        self.is_binary_target = is_binary_target
+        self.na_columns = na_columns
+        self.na_step = na_step
+        self.df: pd.DataFrame = pd.DataFrame()
+        self.treatment_name: Union[str, list, None] = None
+        self.target_names: list = []
+        self.info_col_names: Union[list, None] = None
+        self.features_names: list = []
+        self.ate: float = 0.0
+        self.sigma = 1.3
+        self._generate_df()
+
+    def _generate_df(self):
+        """
+        Generates the dataset and predict ate for it
+        """
+        m_with_dummy = self._generate_feature_cols()
+        info = self._generate_info_cols()
+        treatment = self._generate_treatment_cols()
+        data = self._generate_outcome_and_full_data(m_with_dummy, treatment)
+        self._set_column_names()
+        self.df = self._set_df(data, info)
+        self._set_nans()
+        self._process_ate()
+
+    def _process_ate(self):
+        pass
+
+    def _generate_feature_cols(self):
+        """
+        Generate columns with main features for causal process
+
+        Returns:
+            m_with_dummy:
+                Array with features
+        """
+        means = np.random.uniform(-1, 1, self.num_features)
+        cov_mat = np.identity(self.num_features)
+        m = np.random.multivariate_normal(means, cov_mat, self.num_records)
+        m_with_dummy = to_categories(m, self.num_features, self.num_discrete_features)
+        return m_with_dummy
+
+    def _generate_info_cols(self):
+        """
+        Generate columns with information features for causal process
+
+        Returns:
+            info:
+                List with features
+        """
+        info = []
+        for i in range(self.num_info_cols):
+            if i % 2 == 0:
+                coefficient = 3 if i <= 2 else i
+                array = list(range(1, self.num_records * coefficient, coefficient))
+                random.shuffle(array)
+                info.append(array)
+            else:
```

Comment:
Let's make the generation of string info col  columns a separate function

Reply:
Some functions like `_generate_string_info_col(length: int)` -> np.array 
And append array after it
It would be better for testing and for view

Reply:
I don't really like i%2, maybe it would be better to do two loops? Divide i in half. 

Go through the first loop i // 2 and create numeric columns, which will be created with the help of the function.

And the second loop from (n - i // 2) and create string columns.